src/sqlfmt/api.py

```diff
@@ -54,6 +54,35 @@ def format_string(source_string: str, mode: Mode) -> str:
     return result
 
 
+def format_markdown_string(source_string: str, mode: Mode) -> str:
+    """
+    Takes a Markdown string and a mode as input, returns the the Markdown string with all of its SQL code blocks formatted.
+    """
+    if mode.no_markdownfmt:
+        return source_string
+
+    from mistletoe import Document
+    from mistletoe.block_token import BlockToken, CodeFence
+    from mistletoe.markdown_renderer import MarkdownRenderer
```

Comment:
Let's wrap this in `try ... except ImportError` and re-raise a `SqlfmtError` to create a nicer error message for users who don't have the extra installed. I know right now we shouldn't hit that codepath due to the Mode property, but would be good  to safeguard against that changing in the future.